### PR TITLE
Improved error reporting for Issue707: when module expects extension par...

### PR DESCRIPTION
**Pull request by sgoodgrove**
I've done a pull request for [Issue707](https://github.com/nodeschool/discussions/issues/707) whereby the parameter for the module file is expected to be prefixed with a ".".

I'm not entirely happy with my solution as I'm not happy with calling mod in a nested way, but if I didn't do this, then the callbacks were async and for some reason this makes `Exercise.prototype.process` in `workshopper-exercise/exercise.js` blow up with the message:

```
C:\.......\npm\node_modules\learnyounode\node_modules\workshopper-exercise\exercise.js:177
    processors[i].call(self, mode, function (err, pass) {
                  ^
TypeError: Cannot call method 'call' of undefined
    at next (C:\Users\stevego\AppData\Roaming\npm\node_modules\learnyounode\node_modules\workshopper-exercise\exercise.js:177:19)
    at C:\Users\stevego\AppData\Roaming\npm\node_modules\learnyounode\node_modules\workshopper-exercise\exercise.js:184:7
    at callback (C:\Users\stevego\AppData\Roaming\npm\node_modules\learnyounode\exercises\make_it_modular\verify.js:27:15)
    at modFileError (C:\Users\stevego\AppData\Roaming\npm\node_modules\learnyounode\exercises\make_it_modular\verify.js:32:5)
    at C:\Users\stevego\AppData\Roaming\npm\node_modules\learnyounode\exercises\make_it_modular\verify.js:133:15
    at C:\Users\stevego\Desktop\learnyounode\Solutions\6 Make It Modular Module.js:18:6
    at Object.oncomplete (fs.js:107:15)
```

### CODE REVIEW

File: exercises/make_it_modular/verify.js [hunk 1]
```diff
@@ -197,4 +218,4 @@ function verify (callback) {
   verifyModuleUsed.call(this, callback)
 }
 
-module.exports = verify
\ No newline at end of file
+module.exports = verify
```
Comment (martinheidegger) [inline exercises/make_it_modular/verify.js]: Line break?

File: exercises/make_it_modular/verify.js [hunk 2]
```diff
@@ -88,6 +88,18 @@ function validateModule (modFile, callback) {
 
     callbackUsed = false
     try {
+	function compareArrays(a, b) {
+	    var a2 = (a != null) ? a.slice().sort() : new Array(),
+	    b2 = (b != null) ? b.slice().sort() : new Array(),
```
Comment (martinheidegger) [inline exercises/make_it_modular/verify.js]: how about

```JavaScript
if (a.length != b.length) return false;
```

File: exercises/make_it_modular/verify.js (hunk 1 above)
Comment (sgoodgrove) [inline exercises/make_it_modular/verify.js]: I've made some changes to the indentation and added a line break in. Sorry this is my first pull request. I expect there will be warts in here.

File: exercises/make_it_modular/verify.js (hunk 2 above)
Comment (sgoodgrove) [inline exercises/make_it_modular/verify.js]: Good idea. I've made this change. Do changes made in my branch automatically show in the merge request?

**Comment by rvagg** [comment]
thanks for giving it a go @sgoodgrove, I'm glad we got to be your first pull request

My advice though is to try and grok the prevailling code style and follow that, there's nothing more grating than a pull request that's oblivious to code style. It can be tricky, particularly with JavaScript but you eventually become attuned to the variations in style and what to look out for.

Some things I'm immediately noticing are:

* indentation not matching, you should be using 2-space indenting for this
* whitespace required between keywords and parens, like `if (...` and `function (...`
* no semi-colons unless absolutely necessary

The newline thing doesn't bother me so much as long as you don't add an _additional_ one, I switch between Sublime Text and vim so my files either have a newline or no newline at the end of them depending on what I'm using at the time.

The key here is to minimise diff noise so the reviewer(s) can concentrate just on your addition / change and make a judgment on that. Currently there's too much noise to figure out clearly what's being done here.

Also, while I'm not one to get uptight about having too many commits, it can be a good idea to just amend your existing commit(s) and force push. So `git commit -a --amend` to redo your last commit with new changes and `git push origin branch-name --force` to push the new amended commit to your fork branch, then we won't fill up with new commits each time you make a tweak.

**Comment by ralphtheninja** [comment]
@sgoodgrove Hit me up on gitter https://gitter.im/rvagg/learnyounode if you have any questions. I can fork your repo and help you out with the code style and make sure the commits are amended properly.

**Comment by sgoodgrove** [comment]
I've updated some further code style based on suggestions from @ralphtheninja .

I'm not entirely sure how to improve the compareArrays function, but I could possible introduce underscore.js and write a comparison such as

    function compareArrays(a, b) {
      return _.difference(a, b).length == 0 && _.difference(b, a).length == 0
    }

**Comment by ralphtheninja** [comment]
Still some minor things left.

1. Space after function name here https://github.com/sgoodgrove/learnyounode/blob/36b6e6e7f656a748c87934cef8c72683615b924e/exercises/make_it_modular/verify.js#L89
2. Comma first here https://github.com/sgoodgrove/learnyounode/blob/36b6e6e7f656a748c87934cef8c72683615b924e/exercises/make_it_modular/verify.js#L92-L94
3. Space after for here https://github.com/sgoodgrove/learnyounode/blob/36b6e6e7f656a748c87934cef8c72683615b924e/exercises/make_it_modular/verify.js#L96
4. This check could be moved out of the callback https://github.com/sgoodgrove/learnyounode/blob/36b6e6e7f656a748c87934cef8c72683615b924e/exercises/make_it_modular/verify.js#L147-L149 which means you don't have to check for array length.

**Comment by sgoodgrove** [comment]
Sorry, I thought I'd been pushing my updates up to my repo... but it turn out I'd not even been adding the files locally, so my commits/pushes were not doing anything.

**Comment by rvagg** [comment]
sorry about stepping on toes @sgoodgrove, #227 is basically a dupe of this

**Comment by sgoodgrove** [comment]
@rvagg No problem, I'm happy for you to manage this however you want.

**Comment by a0viedo** [comment]
@rvagg can we close this or other work is pending from this?

**Comment by AnshulMalik** [comment]
#227 does the same thing so...

**Pull request closed without merging**